web/plugs/authorization.ex

```diff
@@ -75,12 +75,12 @@ defmodule Cog.Plug.Authorization do
   # in the future. For example, this would allow users to update
   # related data with paths such as `/v1/users/:user_id/profiles/:id`.
   defp self_updating?(opts, conn) do
-    if Keyword.get(opts, :allow_self_updates, false) == true do
-      conn.private.phoenix_action in [:update, :show] and
-      (conn.assigns.user.id == conn.params["id"] or
-       conn.params["id"] == "me")
-    else
-      false
+    case Keyword.get(opts, :allow_self_updates, []) do
```

Comment:
This can just be `Keyword.fetch!(opts, :allow_self_updates)` if you update the value of  `:allow_self_updates` in `opts` to `self_updates` in `init` before returning instead of just `opts`.